### PR TITLE
Mennekes: Add support for the Amtron Compact 2.0

### PR DESCRIPTION
**Pull request by mzanetti**
nymea-plugins-modbus pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

### CODE REVIEW

**Review by t-mon — CHANGES_REQUESTED** [review]

File: mennekes/integrationpluginmennekes.cpp [hunk 1]
```diff
@@ -116,6 +124,36 @@ void IntegrationPluginMennekes::discoverThings(ThingDiscoveryInfo *info)
 
             info->finish(Thing::ThingErrorNoError);
         });
+        discovery->startDiscovery();
+    } else if (info->thingClassId() == amtronCompact20ThingClassId) {
+        AmtronCompact20Discovery *discovery = new AmtronCompact20Discovery(hardwareManager()->modbusRtuResource(), info);
+        connect(discovery, &AmtronCompact20Discovery::discoveryFinished, info, [this, info, discovery](bool modbusMasterAvailable){
```
[the remark this answers is not in the document]
Comment (mzanetti) [reply mennekes/integrationpluginmennekes.cpp:130]: Which check? This is in discoverThings(), not setupThing() and I am not sure which check you mean as there is none...

[the remark this answers is not in the document]
Comment (t-mon) [reply mennekes/integrationpluginmennekes.cpp:130]: You finish the setup only if you successfully discover the wallbox, which means it must be connectable. I'd do that only if it is the initial setup or a reconfigure

File: mennekes/integrationpluginmennekes.cpp [hunk 2]
```diff
@@ -634,6 +753,116 @@ void IntegrationPluginMennekes::setupAmtronHCC3Connection(ThingSetupInfo *info)
...
+            info->finish(Thing::ThingErrorNoError);
+            info->thing()->setStateValue(amtronCompact20CurrentVersionStateTypeId, compact20Connection->firmwareVersion());
+        } else {
+            delete compact20Connection;
+            info->finish(Thing::ThingErrorHardwareNotAvailable);
+        }
+    });
+
+    connect(compact20Connection, &AmtronCompact20ModbusRtuConnection::initializationFinished, thing, [=](bool success){
+        thing->setStateValue(amtronCompact20ConnectedStateTypeId, success);
```
[the remark this answers is not in the document]
Comment (mzanetti) [reply mennekes/integrationpluginmennekes.cpp]: fair enough. changed.

File: mennekes/integrationpluginmennekes.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (t-mon) [reply mennekes/integrationpluginmennekes.cpp:130]: sorry, yes, the comment was ment to be in the setup...clicked the wrong line

**Review by t-mon — APPROVED** [review]

**Comment by mzanetti** [comment]
merged

**Pull request closed without merging**